docs/2fa/developers/workflow.md

```diff
@@ -0,0 +1,27 @@
+# Workflow
+
+![](../../assets/2fa/workflow.png)
+
+### Key Components
+* Primary wallet:
+
+  The place where users store and control their assets.
+
+* Recovery wallet:
+
+  Used for identifying the users' roles and resetting the 2FA device, the relationship between the primary wallet and the recovery wallet is saved on-chain after binding.
+
+* 2FA client:
+
+  The device which provides the TOTP(Time-based one-time password).
+
+<!-- * [Automata 2FA Guru](https://app.2fa.guru) -->
+* Automata 2FA Guru
+
+  The dApp hosted by the Automata team. It contains the 2FA management panel and wallet assets control panel. Users can sign up 2FA and find the 2FA protected dApps in the management panel, and they can make the authorised transfer under the wallet assets control panel.
```

Comment:
The dApp hosted by the Automata team. It contains the 2FA management and wallet assets control panels. Users can sign up for 2FA and find the 2FA protected dApps in the management panel. They can also make authorised transfers under the wallet assets control panel.